src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -168,7 +168,8 @@ protected ExecutionStrategy(DataFetcherExceptionHandler dataFetcherExceptionHand
                 .thenCompose((fetchedValue) ->
                         completeField(executionContext, parameters, fetchedValue));
 
-        result.whenComplete(fieldCtx::onEnd);
+        fieldCtx.onDispatched(result);
+        result.whenComplete(fieldCtx::onCompleted);
```

Comment:
This pattern of `onDispatched` with a CF passed in and then `cf.whenComplete()` with a value is everywhere